### DISCUSSION TITLE
Typescript Strict mode compatibility

### DISCUSSION THREAD

**Discussion by diovannaschell** [prisma/prisma#11805] [Q&A]
Hello, Thanks for all the work with prisma!

I'm trying to update my APIs to use the typescript on Strict mode especially because of the strict null acertions, but enabling this option broke 90% of my code with prisma:

```ts
const { _sum: total} = await connection.nfe.aggregate({
            _sum: { value: true },
            where: {
                    deleted: 'false', // This line gives "Type 'string' is not assignable to type 'undefined'.ts(2322)"
                    events: {
                       location: { id: { in: locationsIds } }
                    } // An this "Type '{ location: { id:  { in: number[] } }  }' is not assignable to type 'undefined'.ts(2322)"
                },
                status: 'received',
            },
        });
```
This is just one example, but almost every where condition in my code now prompts this errors

this is how vsCode shows the `deleted` property on mouse hover `(property) deleted?: string | Prisma.StringNullableFilter | null | undefined`

Am I missing something? Typescript Strict mode can be used with prisma?

Thanks in advance

**Comment by ovidb** [comment]
@diovannaschell have you ended up find a somewhat elegant solution for this? I was thinking of use some type guards but just wondering if you found something

**Comment by divmgl** [comment]
+1 on this. This is the first time I experience this problem and I'd really like to avoid turning off strict mode. What's going on here? Clearly the type accepts `string`, so why can't I assign a `string`? Really strange.

**Comment by SwenenzY** [comment]
solved?

**Comment by nick-allen** [comment]
Anyone else running into this?

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
